packages/hawtio/src/plugins/camel/endpoints/endpoints-service.ts

```diff
@@ -0,0 +1,128 @@
+import { isObject } from '@hawtiosrc/util/objects'
+
+export class Endpoint {
+  constructor(public uri: string, public state: string, public mbean: string) {}
+}
+
+export async function getEndpoints(node: MBeanNode): Promise<Endpoint[]> {
+  const endpoints: Endpoint[] = []
+  const ctxNode = ccs.findContext(node)
+  if (!ctxNode || ctxNode.childCount() === 0) return endpoints
+
+  const endPointsNode = ctxNode.get(endpointsType) as MBeanNode
+  for (const ep of endPointsNode.getChildren()) {
+    if (!ep.objectName) continue
+    const attributes = await jolokiaService.readAttributes(ep.objectName)
+    endpoints.push(new Endpoint(attributes.EndpointUri as string, attributes.State as string, ep.objectName))
+  }
+
+  return endpoints
+}
+
+export function canCreateEndpoints(node: MBeanNode | null): boolean {
+  return node ? workspace.hasInvokeRights(node, 'createEndpoint') : false
+}
+
+export async function componentNames(node: MBeanNode): Promise<string[]> {
+  const ctxNode = ccs.findContext(node)
+  if (!ctxNode || ctxNode.childCount() === 0 || !ctxNode.objectName) return []
+
+  const names = await jolokiaService.execute(ctxNode.objectName, 'componentNames')
+  return names as string[]
+}
+
+function notifyError(msg: string) {
+  eventService.notify({
+    type: 'danger',
+    message: msg,
+  })
+}
+
+export async function createEndpoint(node: MBeanNode, name: string) {
+  const ctxNode = ccs.findContext(node)
+  if (!ctxNode) {
+    notifyError('Could not find the CamelContext!')
+    return
+  }
+
+  if (!ctxNode.objectName) {
+    notifyError('Could not find the CamelContext MBean!')
+    return
+  }
+
+  jolokiaService
+    .execute(ctxNode.objectName, 'createEndpoint(java.lang.String)', [name])
+    .then((value: unknown) => {
+      if (value === true) {
+        workspace.refreshTree()
+        eventService.notify({
+          type: 'success',
+          message: 'Creating endpoint',
+          duration: 3000,
+        })
+      } else {
+        eventService.notify({
+          type: 'danger',
+          message: 'Invalid URI',
+        })
+      }
+    })
+    .catch((error: string) => {
+      error = error.replace('org.apache.camel.ResolveEndpointFailedException : ', '')
+      eventService.notify({
+        type: 'danger',
+        message: error,
+      })
+    })
+}
+
+export function createEndpointFromData(
+  node: MBeanNode,
+  componentName: string,
+  endPointPath: string,
+  parameters: Record<string, string>,
+) {
+  if (!componentName) console.error('createEndpointFromData: component name must be defined')
+
+  if (!endPointPath) console.error('createEndpointFromData: endpoint path must be defined')
+
+  let uri = componentName + '://' + endPointPath
+
+  log.debug('Have endpoint data ' + JSON.stringify(parameters))
+
+  let params = ''
+  Object.entries(parameters).forEach(([key, value]) => {
+    const prefix = params ? '&' : ''
+    params += prefix + key + '=' + value
```

Comment:
I think this can be put as 

& + Object.entries(parameters).map(entry => entry.join('='))

![image](https://user-images.githubusercontent.com/17336236/230228594-17452db4-776a-4558-8a89-b1a77bb772e2.png)

Reply:
So I believe you can do something like

```
const uri = componentName + '://' + endPointPath
              + params 
              ? '&' + Object.entries(parameters).map(entry => entry.join('=')).join('&')
              : ''

createEndpoint(node, uri)
```